waiter/src/waiter/main.clj

```diff
@@ -56,6 +57,10 @@
             (log/error e "Unable to consume request stream"))))
       resp)))
 
+(defn- wrap-request-cookies [handler]
```

Comment:
The issue is that cookies-response natively doesn't handle async responses. We are doing that currently in cookie-support: I could move that here or this to cookie-support for clarity (or write our own `wrap-cookies`)